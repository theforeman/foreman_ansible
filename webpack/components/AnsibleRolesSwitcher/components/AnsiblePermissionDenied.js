import React from 'react';
import { EmptyStatePattern as EmptyState } from 'foremanReact/components/common/EmptyState';

const AnsiblePermissionDenied = (props) => {
  const description = (
    <span>
      { __('You are not authorized to perform this action.') }<br/>
      { __('Please request one of the required permissions listed below from a Foreman administrator:') }<br/>
    </span>
  );

  const doc = (
    <ul className='list-unstyled'>
      <li>{ 'view_ansible_roles' }</li>
    </ul>
  );

  return (<EmptyState iconType='fa'
                      icon='lock'
                      header={__('Permission Denied')}
                      description={description}
                      documentation={doc} />);
};

export default AnsiblePermissionDenied;
