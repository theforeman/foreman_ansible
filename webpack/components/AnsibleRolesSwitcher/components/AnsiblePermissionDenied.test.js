import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';
import AnsiblePermissionDenied from './AnsiblePermissionDenied';

jest.mock('foremanReact/components/common/EmptyState');

describe(
  'AnsiblePermissionDenied',
  () => testComponentSnapshotsWithFixtures(AnsiblePermissionDenied, { 'should render': {} }),
);
