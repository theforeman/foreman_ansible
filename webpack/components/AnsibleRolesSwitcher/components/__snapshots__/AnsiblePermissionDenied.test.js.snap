// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsiblePermissionDenied should render 1`] = `
<EmptyStatePattern
  description={
    <span>
      You are not authorized to perform this action.
      <br />
      Please request one of the required permissions listed below from a Foreman administrator:
      <br />
    </span>
  }
  documentation={
    <ul
      className="list-unstyled"
    >
      <li>
        view_ansible_roles
      </li>
    </ul>
  }
  header="Permission Denied"
  icon="lock"
  iconType="fa"
/>
`;
